src/OpenTelemetry.Api/DistributedContext/DistributedContextEntry.cs

```diff
@@ -96,27 +96,13 @@ public DistributedContextEntry(string key, string value, in EntryMetadata metada
         /// <inheritdoc/>
         public override bool Equals(object o)
         {
-            if (o is DistributedContextEntry that)
-            {
-                return this.Key == that.Key && this.Value == that.Value;
-            }
-
-            return false;
+            return o is DistributedContextEntry that && (this.Key == that.Key && this.Value == that.Value);
         }
 
         /// <inheritdoc/>
         public override string ToString()
         {
-            if (this.Key is null)
-            {
-                return "{}";
-            }
-
-            return nameof(DistributedContextEntry)
-            + "{"
-            + nameof(this.Key) + "=" + this.Key + ", "
-            + nameof(this.Value) + "=" + this.Value
-            + "}";
+            return this.Key is null ? "{}" : $"{nameof(DistributedContextEntry)}{{{nameof(this.Key)}={this.Key}, {nameof(this.Value)}={this.Value}}}";
```

Comment:
I find compressing this method really difficult to read.

Reply:
I understand. 

Albeit, `ToString` is not something that is business-critical and it's a balance between compactness and readability. For me, it seems more "informational" and lots of `+` ops seem more like noise to me.

Reply:
I think it would have been easier to understand if the inline _if_ was broken into two parts:
```chsarp
if (this.Key is null)
{
    return "{}";
}

return $"{nameof(DistributedContextEntry)}{{{nameof(this.Key)}={this.Key}, {nameof(this.Value)}={this.Value}}}";
```

This helps separate the two cases visually. As I say, not a big issue though.